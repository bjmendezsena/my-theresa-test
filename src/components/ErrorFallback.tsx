export const ErrorFallback = () => {
  return <div>Ooops, something went wrong :(</div>;
};
