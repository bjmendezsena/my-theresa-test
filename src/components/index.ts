export * from './BackButton';
export * from './CategoryCarousel';
export * from './ErrorFallback';
export * from './ExampleComponent';
export * from './FavouriteMoviesGrid';
export * from './GenreList';
export * from './Hero';
export * from './Layouts';
export * from './MovieCard';
export * from './MovieInfo';
export * from './MovieMeta';
export * from './MovieOverview';
export * from './MoviePoster';
export * from './WishlistButton';
