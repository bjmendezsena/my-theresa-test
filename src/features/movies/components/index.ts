export * from './BackButton';
export * from './GenreList';
export * from './Hero';
export * from './MovieInfo';
export * from './MovieMeta';
export * from './MovieOverview';
export * from './MoviePoster';
export * from './WishlistButton';
export * from './MovieCard';
