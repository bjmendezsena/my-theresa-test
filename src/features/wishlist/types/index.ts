export type WishlistItem = {
  id: number;
  title: string;
  posterPath: string;
  releaseDate: Date;
  overview: string;
  rating: number;
};
