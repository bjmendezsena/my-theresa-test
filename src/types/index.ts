import { QueryClient } from '@tanstack/react-query';

export type PrefetchFunction = (
  queryClient: QueryClient,
  params?: Record<string, any>
) => Promise<{
  queryClient: QueryClient;
  queryData?: any;
}>;
