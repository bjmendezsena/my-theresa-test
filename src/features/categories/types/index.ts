import { MovieItem } from '@/features/movies';

export type Category = {
  id: number;
  name: string;
  movies: MovieItem[];
};
