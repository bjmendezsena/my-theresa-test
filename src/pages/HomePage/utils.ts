import { PrefetchFunction } from '@/types';
import { getCategoriesQueryOptions, discoverMoviesQueryOptions } from '@/api';

export const prefetchHomeData: PrefetchFunction = async queryClient => {
  await queryClient.prefetchQuery(getCategoriesQueryOptions());

  const queryData = queryClient.getQueryData(
    getCategoriesQueryOptions().queryKey
  );

  const { genres = [] } = queryData || {};

  await Promise.all(
    genres.map(genre =>
      queryClient.prefetchQuery(
        discoverMoviesQueryOptions({ genres: [genre.id] })
      )
    )
  );

  genres.map(genre =>
    queryClient.getQueryData(
      discoverMoviesQueryOptions({ genres: [genre.id] }).queryKey
    )
  );

  return { queryData, queryClient };
};
