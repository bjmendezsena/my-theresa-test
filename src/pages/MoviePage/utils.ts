import { PrefetchFunction } from '@/types';
import { getMovieQueryOptions } from '@/api';

export const prefetchMovieData: PrefetchFunction = async (
  queryClient,
  params
) => {
  const movieId = params?.id || '';

  await queryClient.prefetchQuery(getMovieQueryOptions(movieId));
  const queryData = queryClient.getQueryData(
    getMovieQueryOptions(movieId).queryKey
  );
  return { queryData, queryClient };
};
