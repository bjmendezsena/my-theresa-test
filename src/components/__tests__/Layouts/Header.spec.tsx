import { renderWithProviders, expect, describe, test } from '@/tests/utils';
import { Header } from '@/components';
import { wishlistStore } from '@/store';
import { WishlistItem } from '@/types';

describe(`<${Header.name}/>`, () => {
  const getComponent = () => renderWithProviders(<Header />);
  test('renders correctly', () => {
    const { container } = getComponent();
    expect(container).toMatchSnapshot();
  });

  test('renders wishlist button with 0 count', () => {
    const { getByText } = getComponent();
    expect(getByText('0')).toBeInTheDocument();
  });
  test('renders wishlist button with correct 1 count', () => {
    const item: WishlistItem = {
      id: 1,
      overview: 'overview',
      posterPath: 'path/to/poster.jpg',
      rating: 5,
      releaseDate: '2023-01-01',
      title: 'Test Movie',
    };

    wishlistStore.setState({
      items: [item],
    });

    const { getByText: getByTextUpdated } = getComponent();
    expect(getByTextUpdated('1')).toBeInTheDocument();
  });
});
