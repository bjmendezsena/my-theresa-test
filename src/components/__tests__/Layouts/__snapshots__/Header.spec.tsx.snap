// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header/> > renders correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__container"
    >
      <a
        class="header__brand"
        data-discover="true"
        href="/"
      >
        <div
          class="header__logo"
        >
          <div
            class="header__logo-icon"
          >
            <span
              class="header__logo-film"
            >
              ðŸŽ¬
            </span>
          </div>
          <div
            class="header__logo-text"
          >
            <span
              class="header__logo-name"
            >
              CineScope
            </span>
            <span
              class="header__logo-tagline"
            >
              Movie Discovery
            </span>
          </div>
        </div>
      </a>
      <div
        class="header__actions"
      >
        <a
          class="header__wishlist-btn"
          data-discover="true"
          href="/whishlist"
        >
          <span
            class="header__wishlist-icon"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-heart"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
              />
            </svg>
          </span>
          <span
            class="header__wishlist-count"
          >
            0
          </span>
        </a>
      </div>
    </div>
  </header>
</div>
`;
